leftwm/src/config.rs

```diff
@@ -96,6 +86,46 @@ pub struct WindowHook {
     pub hiding_strategy: Option<WindowHidingStrategy>,
 }
 
+#[derive(Debug, Clone)]
+pub struct SerializableRegex {
+    regex: Regex,
+}
+
+impl SerializableRegex {
+    /// # Panics
+    ///
+    /// Function can only panic if the str used is not a valid regex
+    #[must_use]
+    pub fn new(str: &str) -> Self {
+        SerializableRegex {
+            regex: Regex::new(str)
```

Comment:
Since we can return an Error from deserialize(), should we also do so here and use a chained let in deserialize() to surface the error upwards without a panic?

Reply:
Added in the below commit. :)

Reply:
Awesome, thank you very much! :)